art/estimators/object_detection/pytorch_detection_transformer.py

```diff
@@ -0,0 +1,998 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2023
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements the task specific estimator for DEtection TRansformer (DETR) in PyTorch.
+
+ | Paper link: https://arxiv.org/abs/2005.12872
+"""
+import logging
+from typing import TYPE_CHECKING, Dict, List, Optional, Tuple, Union, Any
+
+import numpy as np
+
+from art.estimators.object_detection.object_detector import ObjectDetectorMixin
+from art.estimators.pytorch import PyTorchEstimator
+
+if TYPE_CHECKING:
+    # pylint: disable=C0412
+    import torch
+
+    from art.defences.postprocessor.postprocessor import Postprocessor
+    from art.defences.preprocessor.preprocessor import Preprocessor
+    from art.utils import CLIP_VALUES_TYPE, PREPROCESSING_TYPE
+
+logger = logging.getLogger(__name__)
+
+
+def box_cxcywh_to_xyxy(x: "torch.Tensor"):
```

Comment:
We need to mention the license of these functions in their docstrings.

Reply:
Added reference to Apache 2.0